Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -964,4 +1049,135 @@ Common::Matrix44 Wiimote::GetTotalTransformation() const
+  m_balance->SetControlExpression(1, "K");
+  m_balance->SetControlExpression(2, "J");
+  m_balance->SetControlExpression(3, "L");
+
+  // Step Off
+  m_options->SetControlExpression(0, "O");
+
+  // Because our defaults use keyboard input, set calibration shape to a square.
+  m_balance->SetCalibrationFromGate(ControllerEmu::SquareStickGate(1.0));
+}
+
+void BalanceBoard::SubDeviceUpdate(const WiimoteEmu::DesiredWiimoteState& target_state)
+{
+  m_ext.Update(target_state.extension);
+}
+
+void BalanceBoard::Reset()
+{
+  WiimoteBase::Reset();
+
+  m_extension_port.AttachExtension(&m_ext);
+  m_ext.Reset();
+}
+
+BalanceBoard::BalanceBoard(const u8 index) : WiimoteBase(index)
+{
+  using Translatability = ControllerEmu::Translatability;
+
+  // Buttons
+  groups.emplace_back(m_buttons = new ControllerEmu::Buttons(BUTTONS_GROUP_NAME));
+  m_buttons->AddInput(Translatability::Translate, BUTTON_POWER_NAME);
+
+  // Balance
+  groups.emplace_back(
+      m_balance = new ControllerEmu::AnalogStick(
+          _trans("Balance"), std::make_unique<ControllerEmu::SquareStickGate>(1.0)));
+
+  // Options
+  groups.emplace_back(m_options = new ControllerEmu::ControlGroup(_trans("Options")));
+  m_options->AddInput(Translatability::Translate, "Step Off");
+
+  m_options->AddSetting(&m_battery_setting,
+                        {_trans("Battery"),
+                         // i18n: The percent symbol.
+                         _trans("%")},
+                        95, 0, 100);
+
+  m_options->AddSetting(&m_weight_setting,
+                        {_trans("Weight"),
+                         // i18n: The abbreviation for kilograms.
+                         _trans("kg")},
+                        BalanceBoardExt::DEFAULT_WEIGHT, 0, 250);
+
+  Reset();
+}
+
+ControllerEmu::ControlGroup* BalanceBoard::GetBalanceBoardGroup(BalanceBoardGroup group) const
+{
+  switch (group)
+  {
+  case BalanceBoardGroup::Buttons:
+    return m_buttons;
+  case BalanceBoardGroup::Options:
+    return m_options;
+  case BalanceBoardGroup::Balance:
+    return m_balance;
+  default:
+    ASSERT(false);
+    return nullptr;
+  }
+}
+
+ButtonData BalanceBoard::GetCurrentlyPressedButtons()
+{
+  const auto lock = GetStateLock();
+
+  ButtonData buttons{};
+  m_buttons->GetState(&buttons.hex, bboard_bitmasks, m_input_override_function);
+
+  return buttons;
+}
+
+// Update buttons in status struct from user input.
+void BalanceBoard::UpdateButtonsStatus(const DesiredWiimoteState& target_state)
+{
+  m_status.buttons.hex = target_state.buttons.hex & BalanceBoard::BUTTON_POWER;
+}
+
+void BalanceBoard::UpdateBatteryStatus(double charge)
+{
+  // Balance Board battery values are higher than that of a Wii Remote.
+  // This is a linear fit for the charge bars in the home menu.
+  m_status.battery = u8(std::lround((charge + 3.53) / 8.52 * 0xff));
+
+  // TODO: Verify this behavior on real hardware.
+  m_status.battery_low = m_status.battery <= 0x69;
```

Comment:
This value matches `REFERENCE_BATTERY`. I remember doing some reverse engineering and not seeing the calibration value being used, but it's been a while and I don't remember my exact methodology. But, it feels like it's probably not a coincidence.